### DISCUSSION TITLE
How youtube player is working just talking about the youtube video player not the application

### DISCUSSION THREAD

**Discussion by Omar-D** [TeamNewPipe/NewPipe#10523] [Questions]
I am trying to make little youtube app but the player is the bigest problem, my father is gonna to break the pc please
just help me and i will pray to everyone who helped me, thanks in advance ;(

**Comment by opusforlife2** [comment]
It's called ExoPlayer. Check out Just Video Player from F-Droid, which is based on it.

**Reply by Omar-D** [reply]
thanks for answer brother,
i know that my dear bro but i need the source code or the piece code of player part in newpipe source code

**Reply by opusforlife2** [reply]
A simple internet search will give you a link to ExoPlayer's Github repo. It's developed by Google themselves.

**Reply by Omar-D** [reply]
i know that bro,
i have already tried exoplayer before but it's not loading the video right
it's still buffering after 12 second as i remember but newpipe player is very different
it's playing without interrupt,
thanks for everything my dear brother

**Reply by opusforlife2** [reply]
> it's still buffering after 12 second

If your issue is buffering then it's likely a problem with the Youtube-related parts of your app. For that, you need to check out the Extractor repo: https://github.com/TeamNewPipe/NewPipeExtractor

**Reply by Omar-D** [reply]
have you made something similar to this task before,
if yes then please just give me the link to your work
if no then really my laptop is gonna broken

Thanks everytime for you

**Reply by opusforlife2** [reply]
I'm not a developer, sorry.

**Reply by Omar-D** [reply]
ow, your not developer but you have nice informations
thanks my dear bro again,
i hope you have nice day
